src/sonic-config-engine/tests/test_j2files.py

```diff
@@ -319,6 +323,78 @@ def test_qos_and_buffer_arista7800r3_48cq2_lc_render_template(self):
 
+    def do_test_qos_and_buffer_arista7800r3_36dm2_lc_render_template(self, platform, hwsku, minigraph, qos_sample_output, buffer_sample_output):
+        arista_dir_path = os.path.join(self.test_dir, '..', '..', '..', 'device', 'arista', platform, hwsku)
+        # for asic0
+        arista_dir_path = os.path.join(arista_dir_path, '0')
+        qos_file = os.path.join(arista_dir_path, 'qos.json.j2')
+        buffer_file = os.path.join(arista_dir_path, 'buffers.json.j2')
+        port_config_ini_file = os.path.join(arista_dir_path, 'port_config.ini')
+
+        # copy qos_config.j2 and buffer_config.j2 to the Arista7800R3-36DM2-LC directory to have all templates in one directory
+        qos_config_file = os.path.join(self.test_dir, '..', '..', '..', 'files', 'build_templates', 'qos_config.j2')
+        shutil.copy2(qos_config_file, arista_dir_path)
+        buffer_config_file = os.path.join(self.test_dir, '..', '..', '..', 'files', 'build_templates', 'buffers_config.j2')
+        shutil.copy2(buffer_config_file, arista_dir_path)
+
+        for template_file, cfg_file, sample_output_file in [(qos_file, 'qos_config.j2', qos_sample_output),
+                                                            (buffer_file, 'buffers_config.j2', buffer_sample_output) ]:
+            argument = ['-m', minigraph, '-p', port_config_ini_file, '-t', template_file]
+            self.run_script(argument, output_file=self.output_file)
+
+            # cleanup
+            cfg_file_new = os.path.join(arista_dir_path, cfg_file)
+            os.remove(cfg_file_new)
+
+            sample_output_file = os.path.join(self.test_dir, 'sample_output', utils.PYvX_DIR, sample_output_file)
+            assert utils.cmp(sample_output_file, self.output_file), self.run_diff(sample_output_file, self.output_file)
+
+    def test_qos_and_buffer_arista7800r3a_36dm2_c36_render_template(self):
+        self.do_test_qos_and_buffer_arista7800r3_36dm2_lc_render_template('x86_64-arista_7800r3a_36dm2_lc', 'Arista-7800R3A-36DM2-C36',\
+                                                                    self.arista7800r3a_36dm2_c36_lc_t2_minigraph, 'qos-arista7800r3a-36dm2-c36-lc.json',\
+                                                                    'buffer-arista7800r3a-36dm2-c36-lc.json')
+
+    def test_qos_and_buffer_arista7800r3a_36dm2_d36_render_template(self):
+        self.do_test_qos_and_buffer_arista7800r3_36dm2_lc_render_template('x86_64-arista_7800r3a_36dm2_lc', 'Arista-7800R3A-36DM2-D36',\
+                                                                    self.arista7800r3a_36dm2_d36_lc_t2_minigraph, 'qos-arista7800r3a-36dm2-d36-lc.json',\
+                                                                    'buffer-arista7800r3a-36dm2-d36-lc.json')
+
+    def do_test_qos_and_buffer_nokia_ixr7250e_render_template(self, platform, hwsku, minigraph, qos_sample_output, buffer_sample_output):
```

Comment:
why do we need this function? This looks the same as do_test_qos_and_buffer_arista7800r3_36dm2_lc_render_template(), we can pass in the platform to that function and make it generic so that it can be reused

Reply:
done